src/cookies.ts

```diff
@@ -0,0 +1,137 @@
+const memoizedCookies: Map<string, string> = new Map<string, string>();
+
+export const getCookieValues = (name: string): string[] => {
+	const nameEq = `${name}=`;
+	const cookies = document.cookie.split(';');
+
+	return cookies.reduce((acc: string[], cookie: string) => {
+		const cookieTrimmed: string = cookie.trim();
+		if (cookieTrimmed.startsWith(nameEq)) {
+			acc.push(
+				cookieTrimmed.substring(nameEq.length, cookieTrimmed.length),
+			);
+		}
+
+		return acc;
+	}, []);
+};
+
+// subset of https://github.com/guzzle/guzzle/pull/1131
+const isValidCookieValue = (name: string) =>
+	!/[()<>@,;"\\/[\]?={} \t]/g.test(name);
+
+const getShortDomain = ({ isCrossSubdomain = false } = {}) => {
+	const domain = document.domain || '';
+
+	if (domain === 'localhost' || window.guardian?.config?.page?.isPreview) {
+		return domain;
+	}
+
+	// Trim any possible subdomain (will be shared with supporter, identity, etc)
+	if (isCrossSubdomain) {
+		return ['', ...domain.split('.').slice(-2)].join('.');
+	}
+	// Trim subdomains for prod (www.theguardian), code (m.code.dev-theguardian) and dev (dev.theguardian, m.thegulocal)
+	return domain.replace(/^(www|m\.code|dev|m)\./, '.');
+};
+
+const getDomainAttribute = ({ isCrossSubdomain = false } = {}) => {
+	const shortDomain = getShortDomain({ isCrossSubdomain });
+	return shortDomain === 'localhost' ? '' : ` domain=${shortDomain};`;
+};
+
+/**
+ * Set a cookie. If it's been memoized it will replace it's memoized value
+ * @param {string} name - the cookie’s name.
+ * @param {string} value - the cookie’s value.
+ * @param {number} daysToLive - expiry date will be calculated mased on the daysToLive
+ * @param {boolean} isCrossSubdomain - specify if it's a cross subdomain cookie, default false
+ */
+export const setCookie = ({
+	name,
+	value,
+	daysToLive,
+	isCrossSubdomain = false,
+}: {
+	name: string;
+	value: string;
+	daysToLive?: number;
+	isCrossSubdomain?: boolean;
+}): void => {
+	const expires = new Date();
+
+	if (!isValidCookieValue(name) || !isValidCookieValue(value)) {
+		return;
+	}
+
+	if (daysToLive) {
+		expires.setDate(expires.getDate() + daysToLive);
+	} else {
+		expires.setMonth(expires.getMonth() + 5);
+		expires.setDate(1);
+	}
+
+	document.cookie = `${name}=${value}; path=/; expires=${expires.toUTCString()};${getDomainAttribute(
+		{
+			isCrossSubdomain,
+		},
+	)}`;
+
+	// If the cookie is already memoized we want to replace its value
+	if (memoizedCookies.has(name)) {
+		memoizedCookies.set(name, getCookieValues(name)[0]);
+	}
+};
+
+/**
+ * Set a session cookie. If it's been memoized it will replace memoized value
+ * @param {string} name - the cookie’s name.
+ * @param {string} value - the cookie’s value.
+ */
+export const setSessionCookie = (name: string, value: string): void => {
```

Comment:
sorry to be a pain, but do you think it would be clearer if all cookie methods took a consisten type of payload – i.e. a params object like `setCookie`?

Reply:
i feel like i would expect that having done 
```js
setCookie({ name: 'a', value: 'b' })
``` 
i'd expect to be able to do something like this too: 
```js
setSessionCookie({ name: 'a', value: 'b' })
getCookie({ name: 'a', shouldMemoize: false })
```

Reply:
Yes, though so but at that time i though it was simpler to pass params for 1 or 2 values rather than config obj. Is better to be consistent though across the whole module and have refactored it now across all those methods